stl/inc/chrono

```diff
@@ -2086,6 +2087,689 @@ namespace chrono {
+            const utc_time<_Duration>& _Time) noexcept {
+            return gps_time<common_type_t<_Duration, seconds>>{_Time.time_since_epoch()} + _Gps_epoch_adjust;
+        }
+    };
+
+    // [time.clock.file]
+
+    struct _File_time_clock;
+    using file_clock = _File_time_clock;
+
+    template <class _Duration>
+    using file_time = time_point<file_clock, _Duration>;
+#endif // ^^^ _HAS_CXX20
+
+#if _HAS_CXX17
+    inline constexpr long long __std_fs_file_time_epoch_adjustment = 0x19DB1DED53E8000LL; // TRANSITION, ABI
+
+    struct _File_time_clock { // Implementation of trivial-clock
+        using rep                       = long long;
+        using period                    = chrono::system_clock::period;
+        using duration                  = chrono::duration<rep, period>;
+        using time_point                = chrono::time_point<_File_time_clock>;
+        static constexpr bool is_steady = false;
+
+        _NODISCARD static time_point now() noexcept { // get current time; undo epoch adjustment
+            return time_point(duration(_Xtime_get_ticks() + __std_fs_file_time_epoch_adjustment)); // TRANSITION, ABI
+        }
+
+#if _HAS_CXX20
+        // Assumes that FILETIME counts leaps seconds only after the first 27 (i.e., after 1 January 2017), even though
+        // systems can opt out of this behavior.
+        static constexpr seconds _Skipped_filetime_leap_seconds{27};
+        static constexpr sys_days _Cutoff{year_month_day{year{2017}, January, day{1}}};
+
+        template <class _Duration>
+        _NODISCARD static utc_time<common_type_t<_Duration, seconds>> to_utc(const file_time<_Duration>& _File_time) {
+            using _CT = common_type_t<_Duration, seconds>;
+            const auto _Ticks =
+                _File_time.time_since_epoch() - duration_cast<seconds>(duration{__std_fs_file_time_epoch_adjustment});
+
+            if (_Ticks < _Cutoff.time_since_epoch()) {
+                return utc_clock::from_sys(sys_time<_CT>{_Ticks});
+            } else {
+                return utc_time<_CT>{_Ticks + _Skipped_filetime_leap_seconds};
+            }
+        }
+
+        template <class _Duration>
+        _NODISCARD static file_time<common_type_t<_Duration, seconds>> from_utc(const utc_time<_Duration>& _Utc_time) {
+            using _CT = common_type_t<_Duration, seconds>;
+            file_time<_CT> _File_time{duration_cast<seconds>(duration{__std_fs_file_time_epoch_adjustment})};
+
+            if (_Utc_time < utc_seconds{_Cutoff.time_since_epoch()} + _Skipped_filetime_leap_seconds) {
+                _File_time += utc_clock::to_sys(_Utc_time).time_since_epoch();
+            } else {
+                _File_time += _Utc_time.time_since_epoch() - _Skipped_filetime_leap_seconds;
+            }
+
+            return _File_time;
+        }
+#endif // _HAS_CXX20
+    };
+#endif // _HAS_CXX17
+
+#if _HAS_CXX20
+    // [time.clock.conv]
+
+    template <class _Dest_Clock, class _Source_Clock>

```

Comment:
Style: Our template parameters are conventionally `_UglyCamels`, not `_Ugly_Camel_Snakes`. :camel: :snake: (Occurs below.)